website/content/en/docs/best-practices/multi-tenancy.md

```diff
@@ -0,0 +1,131 @@
+linkTitle: "Multi-Tenancy"
+weight: 1
+description: This guide describes the best practices concepts to write Operators for Multi-Tenancy solutions.
+---
+
+## Operator Behavioral Guidelines
+
+### NetworkPolicy 
+
+If your Operator creates or manages [NetworkPolicy][k8s-network-policy] configurations ensure that your solution:
+
+* applies fine-grained network policies to the extent that is required for your managed application to function properly
+* applies fine-grained network policies to enable your managed application internal components to communicate among each other
+* allows users to disable network policy management (your operator doesn't deploy [NetworkPolicy][k8s-network-policy] instances at all)
+* does not create `allow traffic from everywhere in the cluster` type policies
+
+Usage of [NetworkPolicy][k8s-network-policy] is popular in multi-tenant cluster to provide an extra layer
+of segregation among tenants within SDN solutions. Users typically customize this extensively with goal to disallow network traffic among unrelated
+tenants. Operators that deploy a `accept all traffic from anywhere in the cluster` style policy, are creating an obstacle in
+pursuing this goal, especially if it cannot be disabled. In security-conscious environments policies like these are not allowed
+in production. In such cases your operator should minimally **have the option to prevent [NetworkPolicy][k8s-network-policy] objects from being
+created** and leave these to the user. In more advanced cases, your operator should create [NetworkPolicy][k8s-network-policy] configurations that
+follow the least-privilege principle as well implement deny-all-allow-some rules.
+
+### Managing Resources
+
+#### Resource Requests
+
+Create resource requests for CPU and memory:
+
+* for your Operator's manager `Deployment` (See `config/manager/manager.yaml`)
+* for any `Pods` and `Deployments` which are created by your Operator
+
+```yml
+  ...
+  # TODO(user): Configure the resources accordingly based on the project requirements.
+  # More info: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
+  resources:
+    requests:
+      cpu: 10m
+      memory: 64Mi
+    ... 
+```
+
+In order to prevent tenants from consuming all the resources on a cluster or blocking other workloads 
+from being scheduled, many production clusters will define [ResourceQuota][k8s-resources-quotas] configuration.
+[ResourceQuota][k8s-resources-quotas] configuration in a namespace will cause all pods without 
+[Requests and limits][k8s-requests-limits] for `CPU` and `Memory` to be rejected.
+
+Because OLM-managed Operators do not run on Cluster infra nodes, they will consume paid CPU cores. 
+Users will naturally want to limit the maximum amount of resources these non-workload components can take up. 
+For this reason, if you are integrating with [OLM][olm], you should, **always ship your operator `Deployment` with resource requests amounts for CPU and memory**.
```

Comment:
I would rephrase it: "Cluster administrators will naturally want to limit the amount of resources that can be taken away from their workload by other components". Rationale for  the rephrasing is that if the resources are not consumed by the workload there is no harm of them getting used by the operator rather than getting wasted.

I would also flesh that out as the relation may not be obvious to the readers if they don't know the details of the way resources are managed in Kubernetes. Proposal:
Resource request is what is used by Kubernetes for accounting resource availability and for the scheduler to select a node suitable for the pods.

Reply:
But isn't the term "limits" in the rephrased sentence irritating if the user is in fact asked to set requests?

What about "Cluster administrators will naturally want to provide enough resource requests to their Operators in order to give them sufficient memory resources for safe operations, and avoid that other workloads impact them negatively".

After all requests are a way to give a "safe space" to the workload which is setting the request. As long as the workload stay in this requests bounds it will usually not get under pressure or even evicted. This is specifically relevant in resource constraint envs like SNO.

Reply:
adressed.